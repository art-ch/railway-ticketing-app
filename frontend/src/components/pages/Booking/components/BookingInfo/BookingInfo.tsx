import React from 'react';

import { Booking as BookingType } from '@/models';
import { getFormattedDateTime } from '@/utils/formatters';

export type BookingInfoProps = { bookingInfo: BookingType };

export const BookingInfo = ({ bookingInfo }: BookingInfoProps) => {
  const { seatNumber, status, bookingTime } = bookingInfo;

  return (
    <div>
      <p className="font-bold pb-1">Booking Info</p>

      <table className="w-full">
        <thead>
          <tr className="text-sm font-semibold">
            <th className="text-left">Seat</th>
            <th className="text-left">Status</th>
            <th className="text-left">Booked at</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{seatNumber}</td>
            <td>{status}</td>
            <td>{getFormattedDateTime(new Date(bookingTime))}</td>
          </tr>
        </tbody>
      </table>
    </div>
  );
};
