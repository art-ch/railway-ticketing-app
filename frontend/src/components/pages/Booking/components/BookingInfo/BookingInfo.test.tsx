import { render } from '@testing-library/react';
import { BookingInfo } from './BookingInfo';
import { Booking as BookingType } from 'railway-ticketing-app-sdk';

describe('BookingDataSkeleton component', () => {
  it('should be rendered correctly', () => {
    const bookingInfo = {
      seatNumber: 1,
      status: 'CONFIRMED',
      bookingTime: '2025-03-31T05:05:11Z'
    } as BookingType;

    const { container } = render(<BookingInfo bookingInfo={bookingInfo} />);

    expect(container).toMatchSnapshot();
  });
});
