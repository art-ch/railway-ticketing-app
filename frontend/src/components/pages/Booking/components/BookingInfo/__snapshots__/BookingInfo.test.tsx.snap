// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingDataSkeleton component should be rendered correctly 1`] = `
<div>
  <div>
    <p
      class="font-bold pb-1"
    >
      Booking Info
    </p>
    <table
      class="w-full"
    >
      <thead>
        <tr
          class="text-sm font-semibold"
        >
          <th
            class="text-left"
          >
            Seat
          </th>
          <th
            class="text-left"
          >
            Status
          </th>
          <th
            class="text-left"
          >
            Booked at
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            CONFIRMED
          </td>
          <td>
            2025-03-31 08:05
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
